src/components/ProductionLine.js

```diff
@@ -0,0 +1,60 @@
+import React, { useReducer } from "react";
+import { ProductionLineContainer, Main } from "./styled";
+import { initialState, reducer } from "../store/index";
+import Station from "./Station";
+import { STATUSES } from "../constants/statuses";
+
+const ProductionLine = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const handleAddVehicle = (key) => () =>
+    key === "1" ? dispatch({ type: "ADD_VEHICLE" }) : null;
+
+  const handleMoveVehicle = (key) => () =>
+    dispatch({ type: "MOVE_VEHICLE", current_station: parseInt(key) });
+
+  const handleVehicleStatusChange = (key) => (newStatus) => {
+    dispatch({
+      type: "UPDATE_VEHICLE_STATUS",
+      current_station: parseInt(key),
+      newStatus
+    });
+  };
+
+  const shouldWeShowMoveButton = (key, value) => {
+    if (
+      value.vehicle === null ||
+      key === "4" ||
+      value.vehicle.state === STATUSES.BLOCKED
+    )
+      return false;
+
+    return key === "3" ? true : state[parseInt(key) + 1].vehicle === null;
```

Comment:
💡 This is to allow moving the car onto Delivery Truck as it is not one of the stations and hence doesn't have restriction of having only one car at a time.